diff --git a/node_modules/@scottjgilroy/react-native-vision-camera-v4-pose-detection/android/src/main/java/com/visioncamerav3posedetection/VisionCameraV3PoseDetectionModule.kt b/node_modules/@scottjgilroy/react-native-vision-camera-v4-pose-detection/android/src/main/java/com/visioncamerav3posedetection/VisionCameraV3PoseDetectionModule.kt
index 89f218d..f3dd7aa 100644
--- a/node_modules/@scottjgilroy/react-native-vision-camera-v4-pose-detection/android/src/main/java/com/visioncamerav3posedetection/VisionCameraV3PoseDetectionModule.kt
+++ b/node_modules/@scottjgilroy/react-native-vision-camera-v4-pose-detection/android/src/main/java/com/visioncamerav3posedetection/VisionCameraV3PoseDetectionModule.kt
@@ -88,7 +88,13 @@ class VisionCameraV3PoseDetectionModule(proxy : VisionCameraProxy, options: Map<
       addLandmarkToMap(pose.getPoseLandmark(PoseLandmark.LEFT_MOUTH), "leftMouthPosition")
       addLandmarkToMap(pose.getPoseLandmark(PoseLandmark.RIGHT_MOUTH), "rightMouthPosition")
 }
-      return map.toHashMap()
+      val result = HashMap<String, Any>()
+      map.toHashMap().forEach { (key, value) ->
+        if (value != null) {
+          result[key] = value
+        }
+      }
+      return result
     } catch (e: Exception) {
       throw Exception("Error processing pose detection: $e")
     }
